make/compiler_flags

```diff
@@ -117,10 +117,36 @@ CPPFLAGS_SUNDIALS ?= -DNO_FPRINTF_OUTPUT $(CPPFLAGS_OPTIM_SUNDIALS) $(CXXFLAGS_F
 
 ## setup compiler flags
 CXXFLAGS_LANG ?= -std=c++1y
-#CXXFLAGS_BOOST ?=
 CXXFLAGS_SUNDIALS ?= -pipe $(CXXFLAGS_OPTIM_SUNDIALS) $(CPPFLAGS_FLTO_SUNDIALS)
 #CXXFLAGS_GTEST
 
+ifdef BOOST_INC
```

Comment:
We should remove this variable in favor of just using the `BOOST` variable. There's no need to do this if statement.

Let's chat if you need help understanding how make works. (The doc is really tough to understand, so happy to walk you through it.) There are two things that really stand out that we shouldn't do here:
1. `ifdef BOOST_INC` ... the `?=` is the operator you want to use.
2. `CXXFLAGS`... we're not including the C++ includes in our CXX flags. We've managed to get things consistenly split out by having `INC_*` and separating that from `CXXFLAGS_*`. There really aren't codified rules for make anywhere, but let's follow how make itself uses these variables and their naming convention so we look more standard than just inventing a bunch of stuff that makes it harder to manage than it already is.